### DISCUSSION TITLE
List formulas that are manually installed

### DISCUSSION THREAD

**Discussion by mattcen** [Homebrew/discussions#535] [Everyday usage] [answered]
Is there a way to list which formulas have been manually installed, as opposed to installed automatically as dependencies? I assume, given the presence of `brew autoremove`, that Homebrew keeps a record somewhere of this which packages were only installed as dependencies, but I'm not sure how to access this information.

**Comment by carlocab** [comment, marked as answer]
Try `brew bundle dump`. That creates a `Brewfile` in the current directory that should include all the formulae you've installed on request. You can also do `brew leaves` to see all the installed formulae that are not dependencies of another currently installed formula.

**Reply by mattcen** [reply]
Ah, sweet; that looks like it'll do what I want! Thanks so much for the quick and clear response! 😄

**Comment by carlocab** [comment]
As an update, I think you can now do
```
brew list --installed-on-request
```
